libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -532,9 +537,16 @@ void AP_TECS::_update_height_demand(void)
         // Limit height rate of change
         if ((hgt_dem - _hgt_dem_rate_ltd) > (_climb_rate_limit * _DT)) {
             _hgt_dem_rate_ltd = _hgt_dem_rate_ltd + _climb_rate_limit * _DT;
+            _sink_fraction = 0.0f;
         } else if ((hgt_dem - _hgt_dem_rate_ltd) < (-_sink_rate_limit * _DT)) {
             _hgt_dem_rate_ltd = _hgt_dem_rate_ltd - _sink_rate_limit * _DT;
+            _sink_fraction = 1.0f;
         } else {
+            if (is_negative(hgt_dem - _hgt_dem_rate_ltd)) {
+                _sink_fraction = (hgt_dem - _hgt_dem_rate_ltd) / (-_sink_rate_limit * _DT);
```

Comment:
needs a divide-by-zero check

Reply:
I agree, although I can't see any way that it could be a zero, it _sink_rate_limit is complex enough that I think a check is warranted

Reply:
Fixed.